src/Polly.Shared/Monkey/AsyncInjectFaultSyntax.cs

```diff
@@ -0,0 +1,209 @@
+            if (injectionRate == null) throw new ArgumentNullException(nameof(injectionRate));
+            if (enabled == null) throw new ArgumentNullException(nameof(enabled));
+            
+            return new AsyncInjectOutcomePolicy(faultProvider, injectionRate, enabled);
+        }
+    }
+
+    public partial class MonkeyPolicy
+    {
+        #region Exception Based Faults
+        /// <summary>
+        /// Builds an <see cref="AsyncInjectOutcomePolicy{TResult}"/> which injects a fault if <paramref name="enabled"/> returns true and
+        /// a random number is within range of <paramref name="injectionRate"/>.
+        /// </summary>
+        /// <param name="fault">The fault exception object to throw</param>
+        /// <param name="injectionRate">injection rate between [0, 1]</param>
+        /// <param name="enabled">Lambda to check if this policy is enabled in context free fashion</param>
+        /// <returns>The policy instance.</returns>
+        public static AsyncInjectOutcomePolicy<TResult> InjectFaultAsync<TResult>(
+            Exception fault,
+            Double injectionRate,
+            Func<bool> enabled)
+        {
+            if (enabled == null) throw new ArgumentNullException(nameof(enabled));
+
+            Task<Exception> FaultLambda(Context _, CancellationToken __) => Task.FromResult<Exception>(fault);
+            Task<double> InjectionRateLambda(Context _) => Task.FromResult<Double>(injectionRate);
+            Task<bool> EnabledLambda(Context _) => Task.FromResult<bool>(enabled());
+
+            return new AsyncInjectOutcomePolicy<TResult>((Func<Context, CancellationToken, Task<Exception>>)FaultLambda, InjectionRateLambda, EnabledLambda);
+        }
+
+        /// <summary>
+        /// Builds an <see cref="AsyncInjectOutcomePolicy{TResult}"/> which injects a fault if <paramref name="enabled"/> returns true and
+        /// a random number is within range of <paramref name="injectionRate"/>.
+        /// </summary>
+        /// <param name="fault">The fault exception object to throw</param>
+        /// <param name="injectionRate">injection rate between [0, 1]</param>
+        /// <param name="enabled">Lambda to check if this policy is enabled in current context</param>
+        /// <returns>The policy instance.</returns>
+        public static AsyncInjectOutcomePolicy<TResult> InjectFaultAsync<TResult>(
+            Exception fault,
+            Double injectionRate,
+            Func<Context, Task<bool>> enabled)
+        {
+            if (enabled == null) throw new ArgumentNullException(nameof(enabled));
+
+            Task<Exception> FaultLambda(Context _, CancellationToken __) => Task.FromResult<Exception>(fault);
+            Task<double> InjectionRateLambda(Context _) => Task.FromResult<Double>(injectionRate);
+
+            return new AsyncInjectOutcomePolicy<TResult>((Func<Context, CancellationToken, Task<Exception>>)FaultLambda, InjectionRateLambda, enabled);
```

Comment:
why explicit recasting here?

Reply:
>why explicit recasting here?

Good q. Without it the compiler was giving an ambiguous overload resolution error.